oe_theme.theme

```diff
@@ -521,120 +521,163 @@ function oe_theme_preprocess_field(&$variables, $hook): void {
+ * Generates pagination items for ECL pagination component.
  */
 function oe_theme_preprocess_pager(array &$variables): void {
-  // Get the tags for the first, previous, next and last links.
-  $tags = $variables['pager']['#tags'];
-
-  $variables['ecl_items'] = [];
-
-  // Generate fist link if it is available.
-  if (!empty($variables['items']['first']['href'])) {
-    $first_link = [];
-    $first_link['type'] = 'first';
-    $first_link['aria_label'] = t('Go to first page');
-    $label = t('‹‹ First');
-    if (isset($tags[0])) {
-      $label = $tags[0];
-    }
-    $first_link['link']['link'] = [
-      'path' => $variables['items']['first']['href'],
-      'label' => $label,
-    ];
-    $variables['ecl_items'][] = $first_link;
+  $element = $variables['pager']['#element'];
+  $parameters = $variables['pager']['#parameters'];
+  $route_name = $variables['pager']['#route_name'];
+  $route_parameters = $variables['pager']['#route_parameters'] ?? [];
+  global $pager_page_array, $pager_total;
+  // Maximum page number for this pager.
+  $pager_max = $pager_total[$element];
+  // Nothing to do if there is only one page.
+  if ($pager_max <= 1) {
+    return;
+  }
+  // Max number of pages to display next to the current page (left/right).
+  // The first and last page links are excluded from this count.
+  $pager_offset = 2;
+  // The page we are currently paged to.
+  $pager_current = $pager_page_array[$element] + 1;
+  // First page of the pagination calculated by offset from current.
+  $pager_first = $pager_offset < $pager_current ? $pager_current - $pager_offset : 1;
+  // Last page of the pagination calculated by offset from current.
+  $pager_last = $pager_current + $pager_offset;
+
+  // Remove extra pages if last page is larger than maximum.
+  if ($pager_last > $pager_max) {
+    $pager_last = $pager_max;
   }
 
-  // Generate previous link if it is available.
-  if (!empty($variables['items']['previous']['href'])) {
-    $previous_link = [];
-    $previous_link['type'] = 'previous';
-    $previous_link['aria_label'] = t('Go to previous page');
-    $label = t('‹ Previous');
-    if (isset($tags[1])) {
-      $label = $tags[1];
-    }
-    $previous_link['link']['link'] = [
-      'path' => $variables['items']['previous']['href'],
-      'label' => $label,
+  // Array to collect the pagination items.
+  $ecl_items = [];
+
+  // Prepare navigation link item for previous/next links with defaults.
+  $nav_link = [
+    'link' => [
+      'icon' => [
+        'path' => $variables['ecl_icon_path'],
+        'type' => 'ui',
+        'name' => 'corner-arrow',
+        'size' => 'xs',
+        'transform' => 'rotate-270',
+      ],
+    ],
+  ];
+  // Prepare ellipsis item.
+  $ellipsis = [
+    'type' => 'ellipsis',
+    'label' => '...',
+  ];
+
+  // Add 'Previous' navigation link.
+  if ($pager_current > 1) {
+    $options = [
+      'query' => pager_query_add_page($parameters, $element, $pager_page_array[$element] - 1),
```

Comment:
`pager_query_add_page()` is deprecated function in drupal:8.8.0. Use `\Drupal\Core\Pager\PagerManagerInterface::getUpdatedParameters()` instead.

Reply:
Dedicated followup here: OPENEUROPA-3034